distribution/lib/Standard/Base/0.0.0-dev/src/Random.enso

```diff
@@ -1,99 +1,343 @@
 import project.Any.Any
-import project.Data.Boolean.Boolean
-import project.Data.Json.JS_Object
-import project.Data.Numbers.Float
-import project.Data.Numbers.Integer
+import project.Data.Array.Array
+import project.Data.Range.Range
 import project.Data.Text.Text
+import project.Data.Time.Date.Date
+import project.Data.Time.Date_Range.Date_Range
+import project.Data.Time.Time_Of_Day.Time_Of_Day
+import project.Data.Time.Time_Period.Time_Period
+import project.Data.Map.Map
+import project.Data.Set.Set
 import project.Data.Vector.Vector
 import project.Error.Error
 import project.Errors.Illegal_Argument.Illegal_Argument
+import project.Nothing.Nothing
+import project.Runtime.Ref.Ref
 import project.System
+from project.Data.Boolean import Boolean, False, True
+from project.Data.Numbers import Float, Integer
+from project.Data.Range.Extensions import all
 
 polyglot java import java.lang.Integer as Java_Integer
 polyglot java import java.util.Random as Java_Random
 polyglot java import java.util.UUID
-polyglot java import org.enso.base.Random_Utils
+polyglot java import org.enso.base.random.Random_Utils
+polyglot java import org.enso.base.random.RandomInstanceHolder
 
-## PRIVATE
-   UNSTABLE
-   Returns a default seed to use for random number generation.
+type Random
+    ## PRIVATE
+       GROUP Random
 
-   The returned seed may differ between each call to this method.
-get_default_seed : Integer
-get_default_seed = System.nano_time
+       Create a new `Random_Generator` with the specified seed.
+
+       Arguments:
+       - seed: The seed used to initialze the `Random_Generator`.
+
+       > Example
+         Create a new `Random_Generator` and use it.
+
+             import Standard.Base.Random.Random
+
+             rng = Random.new_generator 42
+             i = rng.integer 0 10
+    new_generator : Integer -> Random_Generator
+    new_generator (seed:Integer=get_default_seed) = Random_Generator.new seed
+
+    ## TEXT_ONLY
+       GROUP Random
```

Comment:
Have you checked this doesn't show up in CB?

Reply:
I have not, as I cannot run the IDE.